MR Serverless Follow Ups (#17231)
cockroachdb/docs: src/current/cockroachcloud/plan-your-cluster-serverless.md

> ---

Author: because most dedicated docs were created first, most of our serverless vs dedicated page naming is of the structure `page-name` for dedicated and `page-name-serverless` or `serverless-page-name` for serverless. i agree this should be improved but i'd rather open a new pr than change more page names here. this has been on the back of my mind for a while, though, so thank you for bringing it up!

title: Plan a CockroachDB Serverless Cluster
summary: Plan your cluster's configuration.
toc: true
docs_area: deploy
cloud: true
---

{% include cockroachcloud/filter-tabs/plan-your-cluster.md %}

This page describes how resource usage, pricing, and cluster configurations work in {{ site.data.products.serverless }}.

## Request Units

> All cluster activity, including SQL queries, bulk operations, and background jobs, is measured in Request Units, or RUs. RUs are an abstracted metric that represent the size and complexity of requests made to your cluster. [Pricing](https://cockroachlabs.com/pricing) shows cost estimates of some common queries and how they increase with the size and complexity of the query. In addition to queries that you run, Request Units can be consumed by background activity, such as automatic statistics gathering used to optimize your queries or changefeeds connected to an external sink.

Author: Incorporated most of this, but not seeing the pricing chart that I thought they were going to add to the marketing page since we had to remove this info from docs, so I'm going to delete that sentence for now.

> All cluster activity, including SQL queries, bulk operations, and background jobs, is measured in Request Units, or RUs. RUs are an abstracted metric that represent the size and complexity of requests made to your cluster. [Pricing](https://cockroachlabs.com/pricing) shows cost estimates of some common queries and how they increase with the size and complexity of the query. In addition to queries that you run, Request Units can be consumed by background activity, such as automatic statistics gathering used to optimize your queries or changefeeds connected to an external sink.

Reviewer: > [Pricing](https://cockroachlabs.com/pricing) shows cost estimates of some common queries
Wasn't this page deleted as part of this PR? What would this link to?

Author: this link is to the marketing page on the main site, but it looks like they didn't add the query table back in there so i'm going to delete it

Author: Adding RU estimates back in without the prices

Author: Reverting to original link out to pricing page now that marketing is publishing these again

## Pricing

With {{ site.data.products.serverless }}, you are charged for the storage and activity of your cluster. Cluster activity is measured in [Request Units](#request-units); cluster storage is measured in GiB and is based on the total volume of storage used over the billing period. Request Unit consumption scales to zero when your cluster has no activity, so you will only be charged for what you use.

RU and storage consumption is prorated at the following prices:

>   Unit                    | Cost

Author: IMO this info is important to have in the docs and unlikely to change very often. Also not seeing the pricing calculator.

  ------------------------|------
  1M Request Units        | $0.20
  1 GiB storage           | $0.50

You can see your cluster's RU and storage usage on the [**Cluster Overview** page](cluster-overview-page.html).

## Free vs. paid usage

{{ site.data.products.serverless }} clusters scale based on your workload so that you will only pay for what you use beyond the free resources. All {{ site.data.products.db }} organizations are given 50 million RUs and 10 GiB of storage for free each month. Free resources can be spent across all {{ site.data.products.serverless }} clusters in an organization and will appear as a deduction on your monthly invoice. 

> You must [set resource limits](serverless-cluster-management.html#edit-your-resource-limits) if you've already created one free {{ site.data.products.serverless }} cluster. Setting resource limits will allow your cluster to scale to meet your application's needs and maintain a high level of performance. To set your limits, you can either set storage and RU limits individually, or enter a dollar amount that will be split automatically between both resources. You can also choose an unlimited amount of resources to prevent your cluster from ever being throttled or disabled.

Author: great suggestion, ty!

Unlike {{ site.data.products.dedicated }} clusters, {{ site.data.products.serverless-plan }} clusters can have two regions if your goal is only to [survive zone failures](../{{site.versions["stable"]}}/multiregion-overview.html#survive-zone-failures). Clusters must have at least three regions to ensure that data replicated across regions can [survive region failures](../{{site.versions["stable"]}}/multiregion-overview.html#survive-region-failures).

[Databases created](serverless-cluster-management.html#create-a-database) in {{ site.data.products.serverless }} will automatically inherit all of a cluster's regions, so it is not necessary to run [`ALTER DATABASE ... ADD REGION`](../{{site.versions["stable"]}}/alter-database.html#add-region) to configure regions when adding a database to the cluster. To override the default inheritance, you can specify the primary region with the [`CREATE DATABASE <db_name> WITH PRIMARY REGION`](../{{site.versions["stable"]}}/create-database.html) SQL syntax or the [`sql.defaults_primary_region`](../{{site.versions["stable"]}}/cluster-settings.html#setting-sql-defaults-primary-region) setting.

> For multi-region clusters, storage is billed at the same rate, but because data is replicated three times in the primary region and once in each additional region by default, storage costs will be multiplied based on the number of replicas. Write-heavy applications may experience a significant increase in RU consumption because more resources are required to replicate writes across all regions. Read-heavy applications may see a smaller increase in RU consumption because the resources required to read from a single region of a multi-region deployment are comparable to that of a single region deployment. During the [public preview](../{{site.versions["stable"]}}/cockroachdb-feature-availability.html#feature-availability-phases), RU usage for queries that cross regions will not account for inter-region bandwidth.

Reviewer: > storage costs will be multiplied based on the number of replicas
This is misleading, b/c it sounds like if you have 5 replicas, you'd pay 5x the cost. But we include the cost of the first 3 replicas in the base cost, so you'd only pay 5/3 if you go from 3 => 5 replicas.

Author: tried to clarify this, lmk what you think

> For multi-region clusters, storage is billed at the same rate, but because data is replicated three times in the primary region and once in each additional region by default, storage costs will be multiplied based on the number of replicas. Write-heavy applications may experience a significant increase in RU consumption because more resources are required to replicate writes across all regions. Read-heavy applications may see a smaller increase in RU consumption because the resources required to read from a single region of a multi-region deployment are comparable to that of a single region deployment. During the [public preview](../{{site.versions["stable"]}}/cockroachdb-feature-availability.html#feature-availability-phases), RU usage for queries that cross regions will not account for inter-region bandwidth.

Author: We're going to have region-specific cost multipliers for inter-region traffic, but those haven't been implemented yet, so costs are temporarily lower right now.

## Free vs. paid usage

> {{ site.data.products.serverless }} clusters scale based on your workload so that you will only pay for what you use beyond the free resources. All non-contract {{ site.data.products.db }} organizations are given 50 million [Request Units](#request-units) and 10 GiB of storage for free each month. Free resources can be spent across all {{ site.data.products.serverless }} clusters in an organization and will appear as a deduction on your monthly invoice.

Reviewer 2: ```suggestion
{{ site.data.products.serverless }} clusters scale based on your workload so that you will only pay for what you use beyond the free resources. Each non-contract {{ site.data.products.db }} organization is given 50 million [Request Units](#request-units) and 10 GiB of storage for free each month. Free resources can be spent across all {{ site.data.products.serverless }} clusters in an organization and will appear as a deduction on your monthly invoice.
```
